mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -261,6 +261,12 @@ class WidgetAdapter(
         if (!widget.visibility) {
             return false
         }
+        if (widget.type == Widget.Type.Frame) {
+            // Hide frames with no visible children
+            return items
+                .filter { it.parentId == widget.id }
+                .any { it.visibility }
```

Comment:
This should only return false if there's no visible child, and otherwise defer to the lines below. Otherwise a non-enpty frame will be shown even if one of its parents is invisible (even though I know this is a theoretical case).

Reply:
Yes, I saw this issue as well. Since we don't have nested frames, this shouldn't be an issue, but I can change it.

Reply:
Maybe the function should be renamend now? `shouldShowWidget`?